src/provider/rainbowKit/RainbowKitAccount.ts

```diff
@@ -0,0 +1,378 @@
+      const sigBz = Uint8Array.from(Buffer.from(sig, 'hex'))
+
+      return {
+        signed: doc,
+        signature: {
+          pub_key: {
+            type: ETH_SECP256K1_TYPE,
+            value: pubKeyBase64,
+          },
+          // Remove recovery `v` from signature
+          signature: Buffer.from(sigBz.slice(0, -1)).toString('base64'),
+        },
+      }
+    };
+
+    const signAmino = async (_: string, doc: CarbonTx.StdSignDoc) => {
+      const { account_number, msgs, fee, memo, sequence } = doc
+      const sig = await rainbowKit.signEip712(
+        evmHexAddress,
+        account_number,
+        evmChainId,
+        msgs,
+        fee,
+        memo,
+        sequence)
+      const sigBz = Uint8Array.from(Buffer.from(sig, 'hex'))
+
+      return {
+        signed: doc,
+        signature: {
+          pub_key: {
+            type: ETH_SECP256K1_TYPE,
+            value: pubKeyBase64,
+          },
+          // Remove recovery `v` from signature
+          signature: Buffer.from(sigBz.slice(0, -1)).toString('base64'),
+        },
+      }
+    }
+
+    const signLegacyEip712 = async (signerAddress: string, doc: CarbonTx.StdSignDoc) => {
+      const { account_number, chain_id, msgs, fee, memo, sequence } = doc
+
+      // Only MsgMergeAccount will have an Eth address signer, other generic transaction will be cosmos address signer
+      // FeePayer here is only used for legacy EIP-712
+      const feePayer = AminoTypesMap.fromAmino(msgs[0]).typeUrl === TxTypes.MsgMergeAccount ? AddressUtils.ETHAddress.publicKeyToBech32Address(Buffer.from(pubKeyBase64, "base64"), addressOptions) : signerAddress
+
+      const sig = await rainbowKit.signEip712(
+        evmHexAddress,
+        account_number,
+        chain_id,
+        msgs,
+        fee,
+        memo,
+        sequence,
+        feePayer)
+      return {
+        signed: doc,
+        signature: {
+          pub_key: {
+            type: ETH_SECP256K1_TYPE,
+            value: pubKeyBase64,
+          },
+          signature: Buffer.from(sig, 'hex').toString('base64'),
+        },
+        feePayer,
+      }
+    };
+
+    const getAccounts = async () => {
+      const address = await rainbowKit.defaultAccount()
+      return [
+        {
+          // Possible to change to "ethsecp256k1" ?
+          algo: "secp256k1" as Algo,
+          address,
+          pubkey: Uint8Array.from(Buffer.from(pubKeyBase64, 'base64')),
+        },
+      ]
+    }
+
+    const sendEvmTransaction = async (api: CarbonSDK, req: ethers.providers.TransactionRequest): Promise<string> => {
+      try {
+        const request = await populateEvmTransactionDetails(api, req)
+        const response = await rainbowKit!.sendEvmTransaction(request)
+        return response
+      }
+      catch (error) {
+        console.error(error)
+        throw (parseEvmError(error as Error))
+      }
+    }
+
+    return {
+      type: CarbonSignerTypes.BrowserInjected,
```

Comment:
`legacyEip712SignMode` not defined in this object?

Refer to `src/provider/metamask/MetaMask.ts`
<img width="360" alt="Screenshot 2024-05-13 at 3 19 19 PM" src="https://github.com/Switcheo/carbon-js-sdk/assets/65583522/25119bbb-ffab-49f0-bdf9-a6e4c93ae4e0">